packages/paypal-commerce-integration/src/paypal-commerce-alternative-methods/paypal-commerce-ratepay-payment-strategy.ts

```
@@ -0,0 +1,229 @@
+                methodId: payment.methodId,
+                paymentData,
+            });
+        }
+    }
+
+    finalize(): Promise<void> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<void> {
+        const { legalTextContainer } = this.paypalcommerceratepay || {};
+        if (legalTextContainer) {
+            const legalTextContainerElement = document.getElementById(legalTextContainer);
+            legalTextContainerElement?.remove();
+        }
+
+        return Promise.resolve();
+    }
+
+    private normalizeDate(date: BirthDate) {
+        const formattedDate =
+            date.getDate() < 10 ? `0${date.getDate()}` : date.getDate().toString();
+        const formattedMonth =
+            date.getMonth() < 10 ? `0${date.getMonth() + 1}` : date.getMonth().toString();
+
+        return `${date.getFullYear()}-${formattedMonth}-${formattedDate}`;
+    }
+
+    private renderLegalText(legalTextContainerElementId: string, container: string) {
+        const legalTextContainerId = legalTextContainerElementId;
+        const buttonContainerId = container.split('#')[1];
+        const buttonContainer = document.getElementById(buttonContainerId);
+        const buttonContainerParent = buttonContainer?.parentNode;
+        const legalTextContainer = document.createElement('div');
+        legalTextContainer.style.marginBottom = '20px';
+        legalTextContainer.setAttribute('id', legalTextContainerId);
+        buttonContainerParent?.prepend(legalTextContainer);
+        const paypalSdk = this.paypalCommerceIntegrationService.getPayPalSdkOrThrow();
+        const ratePayButton = paypalSdk.Legal({
+            fundingSource: paypalSdk.Legal.FUNDING.PAY_UPON_INVOICE,
+        });
+        const legalTextContainerElement = document.getElementById(legalTextContainerId);
+
+        if (legalTextContainerElement) {
+            ratePayButton.render(`#${legalTextContainerId}`);
+        }
```

Should we throw an error if the legalTextContainer is not defined?

---

added condition